pkg/utils/utils.go

```diff
@@ -298,6 +299,44 @@ func MonitorMounts() {
 	}
 }
 
+// CleanupOnRestart unmounts and detaches the volumes having
+// DeletionTimestamp set and removes finalizers from the
+// corresponding CStorVolumeAttachment CRs
+func CleanupOnRestart() {
+	var (
+		err        error
+		csivolList *apis.CStorVolumeAttachmentList
+	)
+	// Get list of mounted paths present with the node
+	TransitionVolListLock.Lock()
+	defer TransitionVolListLock.Unlock()
+	if csivolList, err = GetVolListForNode(); err != nil {
+		return
+	}
+	for _, Vol := range csivolList.Items {
+		if Vol.DeletionTimestamp == nil {
+			continue
+		}
+		vol := Vol
+		TransitionVolList[vol.Spec.Volume.Name] = apis.CStorVolumeAttachmentStatusUnmountUnderProgress
+		go func(vol *apis.CStorVolumeAttachment) {
```

Comment:
do this really need to be executed in multiple threads? Is this being done considering the case of large number of unattached volumes? Or is this generally the practice with all code in CSI drivers.

Reply:
It is being run in multiple threads so that if one takes time we don't wait. Unmount and detach is usually a slow process. Might take time.

Reply:
Can you add the above into the comment please.